### PR TITLE
Sc 1604

### PR DESCRIPTION
**Pull request by amitec006**

### CODE REVIEW

File: all-actors/src/main/java/org/sunbird/serviceimpl/CertsServiceImpl.java [hunk 1]
```diff
@@ -42,6 +43,17 @@
         headerMap.put("Content-Type", "application/json");
     }
 
+    @Override
+    public Response delete(Request request) throws BaseException {
+        Map<String, Object> certAddReqMap = request.getRequest();
+        Response response = null;
+        if(StringUtils.isNotBlank((String)certAddReqMap.get(JsonKeys.OLD_ID))){
+            response =  CertificateUtil.deleteRecord((String)certAddReqMap.get(JsonKeys.OLD_ID));
+            logger.info("ertsServiceImpl:delete Deleted the record from cert_registry table for id "+certAddReqMap.get(JsonKeys.OLD_ID));
```
Comment (manzarul) [inline all-actors/src/main/java/org/sunbird/serviceimpl/CertsServiceImpl.java]: ertsServiceImpl:delete  -> certsServiceImpl:delete

File: all-actors/src/main/java/org/sunbird/utilities/CertificateUtil.java [hunk 2]
```diff
@@ -52,6 +53,15 @@ public static boolean isIdPresent(String certificateId) {
         }
         return false;
     }
+    public static Response deleteRecord(String id) throws BaseException {
+        Response response = cassandraOperation.deleteRecord(JsonKeys.SUNBIRD,JsonKeys.CERT_REGISTRY,id);
```
Comment (manzarul) [inline all-actors/src/main/java/org/sunbird/utilities/CertificateUtil.java]: I think in certificate deleted we need to make sure oldId and userId combination need to be found.
Otherwsie delete can have problem.

File: all-actors/src/main/java/org/sunbird/actor/CertBackgroundActor.java [hunk 3]
```diff
@@ -31,11 +31,22 @@ public void onReceive(Request request) throws Throwable {
                 add(request);
                 break;
 
+            case "delete_cert_es":
```
Comment (manzarul) [inline all-actors/src/main/java/org/sunbird/actor/CertBackgroundActor.java]: Delete is not a separate call, During add certificate if user pass oldId then for that user provided certificate will be deleted.

File: all-actors/src/main/java/org/sunbird/utilities/CertificateUtil.java (hunk 2 above)
Comment (amitec006) [reply all-actors/src/main/java/org/sunbird/utilities/CertificateUtil.java]: But recepient identifier is not mandatory.

Comment (amitec006) [reply all-actors/src/main/java/org/sunbird/utilities/CertificateUtil.java]: done

File: all-actors/src/main/java/org/sunbird/actor/CertBackgroundActor.java (hunk 3 above)
Comment (amitec006) [reply all-actors/src/main/java/org/sunbird/actor/CertBackgroundActor.java]: for cassandra delete happen along with add , but ES delete will run in background

File: all-actors/src/test/java/org/sunbird/actor/CertificationActorTest.java [hunk 4]
```diff
@@ -155,10 +195,10 @@ public void generateCertificate() throws Exception {
         Response res = testKit.expectMsgClass(Duration.create(10, TimeUnit.SECONDS),Response.class);
         Assert.assertTrue(null != res && res.getResponseCode() == ResponseCode.OK);
     }
-   // @Test
+    //@Test
```
Comment (manzarul) [inline all-actors/src/test/java/org/sunbird/actor/CertificationActorTest.java]: why this is commented. if not required removed it.

File: all-actors/src/test/java/org/sunbird/actor/CertificationActorTest.java [hunk 5]
```diff
@@ -87,13 +99,41 @@ public void beforeTestSetUp() throws Exception {
...
         when(CertificateUtil.getCertificate(Mockito.anyString())).thenReturn(map);
-
+        when(CertVars.getSERVICE_BASE_URL()).thenReturn("service_base_url");
+        when(CertVars.getDOWNLOAD_URI()).thenReturn("download_url");
+        final Future<HttpResponse<JsonNode>> mockedFuture = Mockito.mock(Future.class);
+        when(CertificateUtil.makeAsyncPostCall(Mockito.anyString(),Mockito.anyString(),Mockito.anyMap())).thenReturn(mockedFuture);
+        final HttpResponse<JsonNode> mockedResponse = Mockito.mock(HttpResponse.class);
+        when(mockedFuture.get()).thenReturn(mockedResponse);
+        when(mockedResponse.getStatus()).thenReturn(HttpStatus.SC_OK);
+        //jsonResponse.getBody().getObject().getJSONObject(JsonKeys.RESULT).getString(JsonKeys.SIGNED_URL)
```
Comment (manzarul) [inline all-actors/src/test/java/org/sunbird/actor/CertificationActorTest.java]: remove commented code.

**Review by manzarul — APPROVED** [review]

**Pull request merged**